docs/messaging/nats-integration.md

```diff
@@ -2,110 +2,201 @@
+When .NET Aspire adds a container image to the app host, as shown in the preceding example with the `docker.io/library/nats` image, it creates a new NATS server instance on your local machine. A reference to your NATS server (the `nats` variable) is added to the `ExampleProject`.
+
+The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `"nats"`. For more information, see [Container resource lifecycle](../fundamentals/app-host-overview.md#container-resource-lifecycle).
+
+> [!TIP]
+> If you'd rather connect to an existing NATS server, call <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString*> instead. For more information, see [Reference existing resources](../fundamentals/app-host-overview.md#reference-existing-resources).
+
+### Add NATS server resource with JetStream
+
+To add the [NATS JetStream](https://docs.nats.io/nats-concepts/jetstream) to the NATS server resource, call the <xref:Aspire.Hosting.NatsBuilderExtensions.WithJetStream*> method:
 
 ```csharp
-public class ExampleService(INatsConnection client)
-{
-    // Use client...
-}
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nats = builder.AddNats("nats");
+                  .WithJetStream();
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(nats);
+
+// After adding all resources, run the app...
 ```
 
-## App host usage
+The NATS JetStream functionality provides a built-in persistence engine called JetStream which enables messages to be stored and replayed at a later time. You can optionally provide a `srcMountPath` parameter to specify the path to the JetStream data directory on the host machine (the provided mount path maps to the container's `-sd` argument).
+
+### Add NATS server resource with data volume
+
+To add a data volume to the NATS server resource, call the <xref:Aspire.Hosting.NatsBuilderExtensions.WithDataVolume*> method on the NATS server resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var nats = builder.AddNats("nats");
+                  .WithDataVolume(isReadOnly: false);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(nats);
 
-To model the Nats resource in the app host, install the [Aspire.Hosting.Nats](https://www.nuget.org/packages/Aspire.Hosting.Nats) NuGet package in the [app host](xref:aspire/app-host) project.
+// After adding all resources, run the app...
+```
+
+The data volume is used to persist the NATS server data outside the lifecycle of its container. The data volume is mounted at the `/var/lib/nats` path in the NATS server container and when a `name` parameter isn't provided, the name is generated at random. For more information on data volumes and details on why they're preferred over [bind mounts](#add-nats-server-resource-with-data-bind-mount), see [Docker docs: Volumes](https://docs.docker.com/engine/storage/volumes).
```

Comment:
Nit: feels like a run on sentence. I swapped "name at random" with "provide the parameter" assuming that the more common scenario is a randomly generated name, considering your example didn't provide a name. Correct?

```suggestion
The data volume is used to persist the NATS server data outside the lifecycle of its container. The data volume is mounted at the `/var/lib/nats` path in the NATS server container. A name is generated at random unless you provide a set the `name` parameter. For more information on data volumes and details on why they're preferred over [bind mounts](#add-nats-server-resource-with-data-bind-mount), see [Docker docs: Volumes](https://docs.docker.com/engine/storage/volumes).
```